game-core/src/main/java/games/strategy/util/ObjectUtils.java

```diff
@@ -0,0 +1,24 @@
+package games.strategy.util;
+
+import javax.annotation.Nullable;
+
+/**
+ * A collection of useful methods for working with instances of {@link Object}.
+ */
+public final class ObjectUtils {
+  private ObjectUtils() {}
+
+  /**
+   * Returns {@code true} if both {@code a} and {@code b} refer to the same object or are both {@code null}; otherwise
+   * returns {@code false}.
+   *
+   * <p>
+   * Use this method only when you really need to compare object references for equality. In almost all cases you should
+   * be using {@link Object#equals(Object)}. <strong>Using this method documents your intention that a reference
+   * equality check is required.</strong>
+   * </p>
+   */
+  public static boolean referenceEquals(final @Nullable Object a, final @Nullable Object b) {
```

Comment:
I thought I was going to have to add a `@SuppressWarnings("ReferenceEquality")` on this method, but, apparently, Error Prone recognizes `referenceEquals()` in [the same way it recognizes `equals()`](http://errorprone.info/bugpattern/ReferenceEquality) and considers use of reference equality in such a method acceptable.